lib/_stream_readable.js

```diff
@@ -1215,17 +1215,33 @@ function endReadableNT(state, stream) {
     state.endEmitted = true;
     stream.emit('end');
 
+    if (stream.writable && stream.allowHalfOpen === false) {
+      process.nextTick(endWritableNT, state, stream);
```

Comment:
attention, [`Duplex` allows more data to be written in the same tick](https://github.com/nodejs/node/pull/31806/files#diff-4c74bb09dc4dabfff99ec3bc35181885L139). However, `net.Socket` did *not* allow this before.